### PR TITLE
ADC esp32s2 attenuation fix for DAC pins

### PR DESCRIPTION
**Pull request by P-R-O-C-H-Y**
## Summary
Changed fixed DAC pins which are different for ESP32S2 to predefined DAC pins from SOC. 

## Impact
None.

### CODE REVIEW

**Comment by P-R-O-C-H-Y** [comment]
> Working with S3 too?

S3 does not support DAC.

**Comment by Jason2866** [comment]
Ahh, yes! This https://github.com/espressif/arduino-esp32/blob/esp32-s3-support/cores/esp32/esp32-hal-adc.c#L38-L41 is probably a copy&paste thing

**Comment by P-R-O-C-H-Y** [comment]
The S3 has 20 ADC channels working, but no DAC channel ;)

File: cores/esp32/esp32-hal-adc.c [hunk 1]
```diff
@@ -142,9 +144,9 @@ bool __adcAttachPin(uint8_t pin){
 #endif
     }
 #if CONFIG_IDF_TARGET_ESP32 || CONFIG_IDF_TARGET_ESP32S2
```
Comment (me-no-dev) [inline cores/esp32/esp32-hal-adc.c:146]: while here, let's use the SOC defines te test wether DAC is supported and maybe how many channels

Comment (P-R-O-C-H-Y) [reply cores/esp32/esp32-hal-adc.c]: SOC define used now.
According to how many channels there are always 2 channels when supported.

Comment (me-no-dev) [reply cores/esp32/esp32-hal-adc.c]: It's not given that this will be true in the future. Why not have more?

**Pull request merged**